src/macaron/vsa/vsa.py

```diff
@@ -135,11 +144,42 @@ class VerificationResult(StrEnum):
     PASSED = "PASSED"
 
 
+def get_common_purl_from_artifact_purls(purl_strs: Iterable[str]) -> str | None:
+    """Get a single common PackageURL given some artifact PackageURLs.
+
+    Assumption: A package may have more than one artifact. If each artifact is identified
+    by a PackageURL, these PackageURLs still share the type, namespace, name, and
+    version values. The common PackageURL contains these values.
+    """
+    try:
+        purls = [PackageURL.from_string(_) for _ in purl_strs]
+    except ValueError:
+        return None
+
+    purl_type = purls[0].type
```

Comment:
Are we assuming that `purls` will never be empty?

Reply:
This function currently does not take an empty collection of PURL strings, because it always gets the result from the `get_components_passing_policy` function, which returns `None` if the result is empty.

That said, I also don't think adding a check here will do any harm. It may be even better to do it anyway. See my commit 077adc05a1a77f2299d418f267a85a97ae2b6e37. I also added some unit tests while I was at it.